source/src/microapp/cs_MicroappController.cpp

```diff
@@ -187,21 +187,29 @@ MicroappController::MicroappController() : EventListener(), _callCounter(0), _mi
 
 	for (int i = 0; i < MICROAPP_MAX_PIN_ISR_COUNT; ++i) {
 		_pinIsr[i].pin = 0;
+		_pinIsr[i].registered = false;
 	}
 	for (int i = 0; i < MICROAPP_MAX_BLE_ISR_COUNT; ++i) {
 		_bleIsr[i].id = 0;
+		_bleIsr[i].registered = false;
 	}
+	_meshIsr.id = 0;
+	_meshIsr.registered = false;
 
 	LOGi("Microapp end is at %p", microappRamSection._end);
 }
 
 /*
- * Get from digital pin to interrupt. See also interruptToDigitalPin.
+ * Get from digital pin to interrupt.
  */
 int MicroappController::digitalPinToInterrupt(int pin) {
 	return pin;
 }
 
+bool MicroappController::softInterruptInProgress() {
```

Comment:
I thought about this as well. I considered resetting the `_emptySoftInterrupts` every tick for example. What I added in this PR is that `_emptySoftInterrupts` is incremented upon a `CS_MICROAPP_COMMAND_SOFT_INTERRUPT_SUCCESS` or a `CS_MICROAPP_COMMAND_SOFT_INTERRUPT_ERROR` since these indicate a softInterrupt has finished and a slot is freed. 

It's hard to say for certain that this it will never get stuck in a state where `_emptySoftInterrupts` is `0`, but I have never encountered it in the tests I've performed.